jvm/src/test/kotlin/app/cash/security_sdk/CertificateTest.kt

```diff
@@ -0,0 +1,102 @@
+  private lateinit var ed25519PrivateKeysetHandle: KeysetHandle
+  private lateinit var p256PrivateKeysetHandle: KeysetHandle
+  private lateinit var ed25519ContentSigner: TinkContentSigner
+  private lateinit var p256ContentSigner: TinkContentSigner
+
+  @BeforeEach
+  fun setUp() {
+    SignatureConfig.register()
+    ed25519PrivateKeysetHandle = KeysetHandle.generateNew(KeyTemplates.get("ED25519"))
+    p256PrivateKeysetHandle = KeysetHandle.generateNew(KeyTemplates.get("ECDSA_P256"))
+
+    ed25519ContentSigner = TinkContentSigner(ed25519PrivateKeysetHandle)
+    p256ContentSigner = TinkContentSigner(p256PrivateKeysetHandle)
+  }
+
+  @Test
+  fun `test signCertificate validity period`() {
+    val certHolder =
+      X509CertificateHolder(createTestMobileSignedCert().certificate)
+    val duration =
+      Duration.ofMillis(
+        certHolder.notAfter.toInstant().minusMillis(certHolder.notBefore.toInstant().toEpochMilli())
+          .toEpochMilli()
+      )
+    assertEquals(180, duration.toDays())
+  }
+
+  @Test
+  fun `test signCertificate certificate entity`() {
+    // Extract cert and then entity value.
+    val certHolder = X509CertificateHolder(
+      createTestMobileSignedCert().certificate
+    )
+    assertEquals("CN=entity", certHolder.subject.toString())
+  }
+
+  @Test
+  fun `test signCertificate certificate issuer`() {
+    // Extract cert and then entity value.
+    val certHolder = X509CertificateHolder(
+      createTestMobileSignedCert().certificate
+    )
+    assertEquals("CN=issuingEntity", certHolder.issuer.toString())
+  }
+
+  @Test
+  fun `test signCertificate signing`() {
+    // Create local copy of issuingCert for use in verifying signature.
+    val issuingCert = CertificateUtil.createRootSigningCertificate(
+      "issuingEntity", Period.ofDays(1), ed25519ContentSigner
+    )
+
+    // Extract the x.509 certificate from our object.
+    val certHolder = X509CertificateHolder(createTestMobileSignedCert().certificate)
+    val issuingCertHolder = X509CertificateHolder(issuingCert.certificate)
+
+    // Cert should verify when presented with issuing cert
+    assertTrue(
+      certHolder.isSignatureValid(
+        TinkContentVerifierProvider(issuingCertHolder.subjectPublicKeyInfo)
+      )
+    )
+
+    // Different key should not verify (in this case it is not self-signed)
+    assertFalse(
+      certHolder.isSignatureValid(
+        TinkContentVerifierProvider(certHolder.subjectPublicKeyInfo)
+      )
+    )
+  }
+
+  private fun createTestMobileSignedCert(): Certificate {
+    val issuingCert = CertificateUtil.createRootSigningCertificate(
+      "issuingEntity", Period.ofDays(1), ed25519ContentSigner
+    )
+
+    val certRequest = CertificateUtil.mobileCertRequestToCertRequest(
+      CertificateUtil.createMobileCertRequest(
+        "entity",
+        p256ContentSigner
+      ).encodeByteString()
+    )
+
+    return issuingCert.signCertificate(certRequest, ed25519ContentSigner)
```

Comment:
this is probably the point I was driving at, we could potentially instantiate some random certificate, but all we care about is the parameters which are the cert request and content signer (issuer in this case).

Reply:
```
    val someOtherCert = CertificateUtil.createRootSigningCertificate(
      "someOtherCert", Period.ofDays(1), ed25519ContentSigner
    )
    
    someOtherCert.signCertificate(certRequest, ed25519ContentSigner)
```

this will result in the same operation. the question becomes why do we need to instantiate a certificate in order to issue a signed cert from the CA

Reply:
generally certs have an 'issued by' field, which if we're using the BC lib needs to be populated (in our case right now we're just using the subjectname)